Don't check for var._obj in template._clean_data

alikins opened the pull request:
AnsibleUnsafe or other unsafe vars used to have a
'_obj' slot but no longer do. This was causing attribute
errors if a object was 'unsafe' but not a string.

##### ISSUE TYPE
<!--- Pick one below and delete the rest: -->

 - Bugfix Pull Request


##### COMPONENT NAME
<!--- Name of the module/plugin/module/task -->
lib/ansible/template/__init__.py

##### ANSIBLE VERSION
<!--- Paste verbatim output from “ansible --version” between quotes below -->
```
ansible 2.3.0 (template_clean_data_obj_fix 2db47257cd) last updated 2017/01/31 11:31:48 (GMT -400)
  config file = /home/adrian/.ansible.cfg
  configured module search path = [u'/home/adrian/ansible/my-modules', u'/home/adrian/ansible/lib/modules/']
```

##### SUMMARY

alikins committed:
Don't check for var._obj in template._clean_data

AnsibleUnsafe or other unsafe vars used to have a
'_obj' slot but no longer do. This was causing attribute
errors if a object was 'unsafe' but not a string.

alikins committed:
More unit tests for test_templar

Add tests for AnsibleUnsafe, lookups, and AnsibleContext

mattclay commented on test/units/template/test_templar.py:
```diff
@@ -111,3 +319,143 @@ def test_template_jinja2_extensions(self):
+                                           {'an_unsafe_dict': wrap_var('some unsafe string 1')}
+                                           })
+        res = context.resolve('some_unsafe_key')
+        self.assertTrue(self.is_unsafe(res['an_unsafe_dict']), 'return of AnsibleContext.resolve (%s) was expected to be marked unsafe but was not' % res['an_unsafe_dict'])
```
CI failure due to PEP 8 issue:

```
2017-01-31 22:24:00 ERROR: PEP 8: test/units/template/test_templar.py:447:161: E501 line too long (172 > 160 characters) (current)
```

abadger commented on lib/ansible/template/__init__.py:
```diff
@@ -348,7 +348,7 @@ def template(self, variable, convert_bare=False, preserve_trailing_newlines=True
                 # Do we need to convert these into text_type as well?
                 # return self._clean_data(to_text(variable._obj, nonstring='passthru'))
-                rval = self._clean_data(variable._obj)
+                rval = self._clean_data(variable)
```
This looks like we should now call _clean_data(variable) unconditionally, no need for the if isinstance check anymore.

abadger commented:
Pointed out one logic simplification here.  I don't know the templating or UNSAFE proxy code so I'll assign this to jimi-c for final approval.

alikins committed:
Always pass vars to _clean_data in templar

_clean_data() will veryify the type is compatible.

update tests to reflect this doesnt throw exceptions
now.
